source/configuration_manual/nfs.rst

```diff
@@ -4,178 +4,109 @@
+  the users between backends more expensive though. It also needs a way to
+  delete the list indexes for users that have already moved to different
+  backends.
+* Use ``mail_location = ...:INDEX=/fast/%2.256Nu/%u:ITERINDEX`` to use
+  "smaller fast storage" for index files and "larger slow storage" for mail
+  files. The ``ITERINDEX`` is used to list mailboxes via the fast index
+  storage rather than the slow mail storage.
+* Use ``mail_location = ...:ALT=/slow/%2.256Nu/%u:NOALTCHECK`` to use
+  "smaller fast storage" for new mails and "larger slow storage" for old
+  mails. The ``doveadm altmove`` command needs to be run periodically. The
+  ``NOALTCHECK`` disables a sanity check to make sure alt storage path doesn't
+  unexpectedly change.
+* Slightly unsafe: Use ``nolock`` mount option to keep all locking within the
+  server. Assuming director works perfectly, there is no need to use locking
+  across NFS. Each user only locks their own files, and the user should only
+  be accessed by a single server at a time. Unfortunately, this doesn't work
+  100% of the time so in some rare situations the same user can become
+  accessed by multiple servers simultaneously. In those situations the mails
+  are more likely to become corrupted if ``nolock`` is used.
 
+Clock synchronization
+=====================
 
-.. note::
+Run ntpd in the NFS server and all the NFS clients to make sure their
+clocks are synchronized. They should always be less than 1 second apart
+from each others.
 
-   The only way to reliably implement the 2nd setup is with :ref:`director <dovecot_director>` service.
+Clustering without director
+===========================
 
-Dovecot configuration
-=====================
+Some people are using Dovecot in a cluster without the director service.
+Below are some suggestions for improving the reliability of this
+configuration.
 
-Single Dovecot server setup or :ref:`Dovecot director <dovecot_director>` cluster setup:
+.. warning:: This method is almost guaranteed to give random errors and can
+             potentially lose emails.
 
-::
+* Configure load balancer so that connections from the same source IP are
+  redirected to the same Dovecot server. This way a single client using
+  multiple IMAP connections doesn't immediately cause problems.
 
-   mmap_disable = yes
-   #dotlock_use_excl = no # only needed with NFSv2, NFSv3+ supports O_EXCL and it's faster
-   mail_fsync = always
-   mail_nfs_storage = no
-   mail_nfs_index = no
+* Set :ref:`setting-mail_nfs_index` = ``yes`` and
+  :ref:`setting-mail_nfs_storage` = ``yes``. These will attempt to flush the NFS
+  caches at appropriate times. However, it doesn't work perfectly.
 
-Multi-server setup that **tries** to flush NFS caches (increases NFS
-operations, and **isn't fully reliable**), try not to use this:
+    * Disabling NFS attribute cache helps a lot in getting rid of caching
+      related errors, but this makes the performance MUCH worse and increases
```

Comment:
Not sure, but I think it is just "performance" without "the". E.g.in the next paragraph it is used that way and sounds nicer.